0.Exploratory_Data_Analysis/nbconverted/Exploratory_Data_Analysis.py

```diff
@@ -0,0 +1,190 @@
+    df_stats(df)
+    # Drop na and reindex accordingly
+    df = df.dropna()
+    df.reindex()
+    # Check for Nans again
+    df_stats(df)
+    # Understand categorical data such as treatment and dosing
+    df[["Metadata_treatment", "Metadata_dose"]].drop_duplicates()
+    # create a list with only columns from the data frame that start with "Metadata"
+
+    treatments = df["Metadata_treatment"].unique()
+    print(treatments)
+
+    n_clusters = len(treatments)
+    # print(n_clusters)
+    samples = str(treatments).strip("[]")
+    # print(samples)
+    new_wave_label = f"{wave_label}_treatments:{samples}"
+
+    subset_number = 1500
+    df_subset = df.sample(n=subset_number)
+    # Code snipptet for metadata extraction by Jenna Tomkinson
+    df_metadata = list(df_subset.columns[df_subset.columns.str.startswith("Metadata")])
+
+    # define which columns are data and which are descriptive
+    df_descriptive = df_subset[df_metadata]
+    df_values = df_subset.drop(columns=df_metadata)
+
+    treatment_ids = df_descriptive["Metadata_treatment"]
+    # Cluster data
+    # clustering code adapted from https://www.kaggle.com/code/aussie84/clustering-with-kmeans-pca-tsne
+
+    kmeans = KMeans(n_clusters=n_clusters)
+    clustering_ori = kmeans.fit_predict(df_values)
+
+    X = df_values
+    # n_components is the number of dimensions to reduce to
+    Xtsne = TSNE(n_components=2).fit_transform(X)
+    dftsneFull = pd.DataFrame(Xtsne)
+
+    dftsneFull["cluster"] = clustering_ori
+    dftsneFull.columns = ["x1", "x2", "cluster"]
+    dftsneFull["Treatment"] = (
+        df_descriptive["Metadata_treatment"].reset_index().drop("index", axis=1)
+    )
+    # Figure Showing tSNE of Clusters vs Treatment
+    fig, ax = plt.subplots(1, 2, figsize=(12, 6))
+    plot = sns.scatterplot(
+        data=dftsneFull,
+        x="x1",
+        y="x2",
+        hue="cluster",
+        legend="full",
+        alpha=0.7,
+        ax=ax[0],
+    )
+    ax[0].set_title("Visualized on TSNE")
+    plot = sns.scatterplot(
+        data=dftsneFull,
+        x="x1",
+        y="x2",
+        hue="Treatment",
+        legend="full",
+        alpha=0.7,
+        ax=ax[1],
+    )
+    ax[1].set_title("Visualized on TSNE")
+    fig.suptitle(
+        f"Comparing Clusters vs Treatment tSNE for {subset_number} subset samples for {samples} treatments"
+    )
+    plt.savefig(f"Figures/tSNE/{new_wave_label}_tSNE_cluster.png")
+    df_values["cluster"] = clustering_ori
+    # callable function for graphing features that contribute most to each cluster's grouping
+    plot_features_all_cluster(
+        df=df_values,
+        label_col="cluster",
+        n_clusters=n_clusters,
+        sensitivity=0.2,
+        file_label=new_wave_label,
+    )
+
+
+# ## Wave1 Dialte50 Analysis
+
+# In[3]:
+
+
+# Import data with low memory arg as the data are large
+df = pd.read_csv(
+    "../../Extracted Features (CSV files)/interstellar_wave1_dilate50_sc.csv.gz",
```

Comment:
When dealing with paths, the lab loves to use `pathlib` library as it inherently knows which os pathing style it should take. . 

Also computers [do not like spaces](https://instr.iastate.libguides.com/datashare/filenames#:~:text=Avoid%20spaces&text=Spaces%20are%20not%20supported%20by,dashes%20(%2D)%20or%20underscores%20(_).) in file names :). Please use underscores instead of spaces